src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,182 @@
+    }
+
+    public void printIncorrectInput(boolean isEmptyDescription, String keyword) {
+        printDividerLine();
+        if (keyword.equalsIgnoreCase("todo")) {
+            System.out.println("\t ☹ OOPS!!! The description of a todo cannot be empty.");
+        } else if (keyword.equalsIgnoreCase("deadline")) {
+            if (isEmptyDescription) {
+                System.out.println("\t ☹ OOPS!!! The description of a deadline cannot be empty.");
+            } else {
+                System.out.println("\t ☹ OOPS!!! The description of a deadline cannot be in the wrong format.");
+            }
+        } else {
+            if (isEmptyDescription) {
+                System.out.println("\t ☹ OOPS!!! The description of an event cannot be empty.");
+            } else {
+                System.out.println("\t ☹ OOPS!!! The description of an event cannot be in the wrong format.");
+            }
+        }
+        printDividerLine();
+    }
+
+    public void addTask(String line) {
+        String[] keywords = line.split(" ", 2);
+        boolean hasNoException = false;
+        boolean isEmptyDescription = true;
+
+        if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("todo")) {
+            try {
+                tasks.add(new Todo(keywords[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(false, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        } else if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("deadline")) {
+            try {
+                String[] details = keywords[1].split(" /by ");
+                isEmptyDescription = false;
+                tasks.add(new Deadline(details[0], details[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(isEmptyDescription, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        } else if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("event")) {
+            try {
+                String[] details = keywords[1].split(" /at ");
+                isEmptyDescription = false;
+                tasks.add(new Event(details[0], details[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(isEmptyDescription, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        }
+        if (hasNoException) {
+            printAddedTasked();
+            updateFile();
+        }
+    }
+
+    public void listTasks() {
+        printDividerLine();
+        System.out.println("\t Here are the tasks in your list:");
+        for (int i = 0; i < Task.getNumberOfTasks(); i++) {
+            System.out.println("      " + (i + 1) + "." + tasks.get(i));
+        }
+        printDividerLine();
+    }
+
+    public void markAsDone(String index) {
+        try {
+            int indexInteger = Integer.parseInt(index);
```

Comment:
Since you are calling this variable index, you can do the -1 in this line itself
`int indexInteger = Integer.parseInt(index) - 1;`